docs/utilities/idempotency.md

```diff
@@ -910,22 +944,38 @@ You can create your own persistent store from scratch by inheriting the `BasePer
 
     For example, the `_put_record` method needs to raise an exception if a non-expired record already exists in the data store with a matching key.
 
-### Modifying the Idempotent Repsonse
-
-The IdempotentConfig allows you to specify a _**response_hook**_ which is a function that will be called when an idempotent response is loaded from the PersistenceStore.
+### Manipulating the Idempotent Response
 
-You can provide the response_hook using _**IdempotentConfig**_.
+You can set up a `response_hook` in the `IdempotentConfig` class to access the returned data when an operation is idempotent. The hook function will be called with the current deserialized response object and the Idempotency record.
 
 === "Using an Idempotent Response Hook"
 
-```python hl_lines="10-15 19"
---8<-- "examples/idempotency/src/working_with_response_hook.py"
-```
+    ```python hl_lines="18 20 23 32"
+    --8<-- "examples/idempotency/src/working_with_response_hook.py"
+    ```
+
+=== "Sample event"
+
+    ```json
+    --8<-- "examples/idempotency/src/working_with_response_hook_payload.json"
+    ```
 
 ???+ info "Info: Using custom de-serialization?"
 
     The response_hook is called after the custom de-serialization so the payload you process will be the de-serialized version.
 
+#### Being a good citizen
+
+Using Response hooks can add subtle improvements to manipulating returned data from idempotent operations, but also add significant complexity if you're not careful.
+
+Keep the following in mind when authoring hooks for Idempotency utility:
+
+1. **Response hook works exclusively when operations are idempotent.** Carefully consider the logic within the `Response hook` and prevent any attempt to access the key from relying exclusively on idempotent operations.
+
+2. **Catch your own exceptions.** Catch and handle known exceptions to your logic.
+
+3. **Watch out when you are decorating the Lambda Handler and using the Response hook.** If you don't catch and handle exceptions in your `Response hook`, your function might not run properly.
```

Comment:
A suggestion to make this more clear:

1- response hooks only work on idempotent responses
2- catch exceptions
3- keep it simple

Is this better?

```suggestion
When using response hooks to manipulate returned data from idempotent operations, it's important to follow best practices to avoid introducing complexity or issues. Keep these guidelines in mind:

1. **Response hook works exclusively when operations are idempotent.** The hook will not be called when an operation is not idempotent, or when the idempotent logic fails.

2. **Catch and Handle Exceptions** Your response hook code should catch and handle any exceptions that may arise from your logic. Unhandled exceptions will cause the Lambda function to fail unexpectedly.

3. **Keep Hook Logic Simple** Response hooks should contain minimal, straightforward logic for manipulating response data. Avoid complex conditional branching and aim for hooks that are easy to reason about.
```

Reply:
This is much better than the one I wrote! Thank you for this very useful review and suggestion @rubenfonseca